### PR TITLE
Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by mechanomi**
this will auto-request a review from the @crate/docs team (but a review from the team is not mandatory)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,3 @@
+*.rst @docs
```
Comment (mfussenegger) [inline .github/CODEOWNERS]: I think this should be `@crate/docs`, at least in other repos we use the `@crate/` prefix.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+blackbox/docs/ @docs
+blackbox/test_docs.py @docs
```
Comment (mfussenegger) [inline .github/CODEOWNERS]: Not sure if you need to get notifications for this?

Comment (mechanomi) [reply .github/CODEOWNERS]: I don't need to. but I figure I'm interested in changes to the docs tests setup. this change does not introduce mandatory review or approval

**Comment by mechanomi** [comment]
@mfussenegger pushed a fixup

File: .github/CODEOWNERS (hunk 2 above)
Comment (seut) [reply .github/CODEOWNERS]: Using the CODEOWNER feature to get notification about file changes one is interested in but not responsible for is abusing it IMHO. Please use something like https://app.github-file-watcher.com/ for this.

Comment (mechanomi) [reply .github/CODEOWNERS]: I would argue that the docs team _is_ responsible for the docs tests. it's a joint responsibility with engineering

**Pull request merged**